projects/components/src/lib/directives/aria-activedescendant/aria-activedescendant.directive.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+import { Directive, ElementRef, Input, Renderer2 } from '@angular/core';
+import { IdGenerator } from '../../../utils';
+import { AriaActivedescendantService } from './aria-activedescendant.service';
+
+@Directive({
+    selector: '[vcdAriaActivedescendant]',
+})
+export class AriaActivedescendantDirective {
+    private readonly uniqueId: string;
+
+    @Input('vcdAriaActivedescendant')
+    public set isActive(value: boolean) {
+        if (value) {
+            this.activedescendantService.selectDescendant(this.uniqueId);
+        }
+    }
+
+    constructor(
+        private elementRef: ElementRef,
+        private renderer: Renderer2,
+        private activedescendantService: AriaActivedescendantService,
+        private idGenerator: IdGenerator
+    ) {
+        this.uniqueId = this.idGenerator.generate();
+        this.renderer.setAttribute(this.elementRef.nativeElement, 'id', this.uniqueId);
```

Comment:
We usually try to avoid making calls from constructor, that's why `ngOnInit` exists.

You could also initialize `uniqueId` in one shot instead of having a separate declaration and an assignment 

`private readonly uniqueId = this.idGenerator.generate();`